src/Search/Gearman/Command/ImportCommand.php

```diff
@@ -165,36 +164,58 @@ public function importBlogArticles()
         $this->iterateSerializeTask($articles, 'blog-article', 'getId', $articles->count());
     }
 
-    private function iterateSerializeTask(Iterator $items, string $type, $method = 'getId', int $count = 0, $skipInvalid = false)
+    private function iterateSerializeTask(Sequence $items, string $type, $method = 'getId', int $count = 0, $skipInvalid = false)
     {
         $this->logger->info("Importing $count items of type $type");
         $progress = new ProgressBar($this->output, $count);
         $limit = $this->limit;
 
-        $items->rewind();
-        while ($items->valid()) {
+        // lazy iterate here instead of relying on the SDK methods
+        foreach ($this->lazySlices($items) as $item) {
             if ($limit->hasBeenReached()) {
                 throw new RuntimeException('Command cannot complete because: '.implode(', ', $limit->getReasons()));
             }
             $progress->advance();
             try {
-                $item = $items->current();
-                if (null === $item) {
-                    $items->next();
-                    continue;
-                }
                 $this->enqueue($type, $item->$method());
             } catch (Throwable $e) {
                 $item = $item ?? null;
                 $this->logger->error('Skipping import on a '.get_class($item), ['exception' => $e]);
                 $this->monitoring->recordException($e, 'Skipping import on a '.get_class($item));
             }
-            $items->next();
         }
         $progress->finish();
         $progress->clear();
     }
 
+    private function lazySlices(Sequence $items): \Generator
+    {
+        $denormalizerProperty = new ReflectionProperty($items, 'denormalizer');
+        $denormalizerProperty->setAccessible(true);
+
+        // create 100-item slices
+        $total = $items->count();
+        $sliceStart = 0;
+        $sliceSize = 100;
+        while ($total >= $sliceStart) {
+            foreach ($items->slice($sliceStart, $sliceSize)->toArray() as $item) {
+                // create a Generator to iterate over items in a slice
+                yield $item;
+            }
+            $sliceStart += $sliceSize;
+
+            // This is the magic memory fixer. Without resetting the serialiser, memory continues to grow
+            $denormalizerProperty->setValue($items, $this->getNewSerializer());
```

Comment:
It would be worth fixing this issue inside `Sequence` (or where ever it is in the PHP SDK), even if that's only to reset the denormalizer after each slice call, but perhaps that should be an issue to address later?

Reply:
Thanks Scott, I'll create a ticket for it.